flink-runtime/src/main/java/org/apache/flink/runtime/rpc/akka/AkkaInvocationHandler.java

```diff
@@ -123,6 +125,21 @@ public Object invoke(Object proxy, Method method, Object[] args) throws Throwabl
 				"retrieve a properly FencedRpcGateway.");
 		} else {
 			result = invokeRpc(method, args);
+			if (result instanceof CompletableFuture) {
+				result = ((CompletableFuture) result).thenApply((Object o) -> {
+					if (o instanceof SerializedValue) {
+						try {
+							return  ((SerializedValue) o).deserializeValue(getClass().getClassLoader());
+						} catch (IOException e) {
+							throw new CompletionException(e);
+						} catch (ClassNotFoundException e) {
+							throw new CompletionException(e);
+						}
+					} else {
+						return o;
+					}
+				});
+			}
```

Comment:
This block should go into the `invokeRpc` method because we also need to unwrap the synchronous calls.

Reply:
We should also add a test case that synchronous calls (Rpc's which return directly a value instead of a `CompletableFuture`) work with this change as well.